### PR TITLE
Make ROOT build also when Alien-Runtime is not there

### PR DESCRIPTION
**Pull request by ktf**
Right now I think it simply works by accident, since the required
dependencies on libxml2 and openssl are very likely to be found on the
system. This should make sure that we build our own version in case
that's not the case.

### CODE REVIEW

**Comment by kouzinopoulos** [comment]
Referring to issue [#905](https://github.com/alisw/alidist/issues/905)

Installing libssl-dev fixed the issue as per the instructions of @sawenzel 

AliDoctor O2 did not report any error  on the missing library dependency

File: defaults-o2-daq.sh [hunk 1]
```diff
@@ -42,6 +42,7 @@ overrides:
       - FreeType:(?!osx)
       - Python-modules
       - libxml2
+      - OpenSSL
       - "GCC-Toolchain:(?!osx)"
```
Comment (sawenzel) [inline defaults-o2-daq.sh]: libxml + openssl also need to be added to `defaults-o2.sh`

Comment (ktf) [reply defaults-o2-daq.sh]: Actually, those should come from `Alien-Runtime` in the case of `default-o2.sh`.

Comment (sawenzel) [reply defaults-o2-daq.sh]: Yes, but it is usually better to be explicit about dependencies ... and not rely in implicit stuff which is hard to track? In any case, if it already works no problem ...

**Comment by awegrzyn** [comment]
Hi, what's the status of this PR?

**Comment by dberzano** [comment]
Let me have a look: I have an independent change in the ROOT recipe to be merged too, I'll open a separate PR so that we can merge them together and avoid too many rebuilds.

**Comment by dberzano** [comment]
OK so at the moment I am refactoring it by making it possible to use ROOT without AliEn-Runtime at all. Some packages, now part of AliEn-Runtime, are in any case going to be taken out after the migration to JAlien. This is the list of packages:

* OpenSSL
* XRootD
* libxml2

**Comment by dberzano** [comment]
I have by the way found an interesting loophole in CODEOWNERS. This PR was opened by @ktf but edited by me (I am an allowed maintainer). Now, I can review _my own_ code just because the PR was originally opened by @ktf :-)

That said, this should address the problem of compiling ROOT without AliEn-Runtime available, and should pave the way for JAlien as well (@costing @mimarpe).

**Comment by dberzano** [comment]
Well: this iteration does not work, at least for ROOT 5 on slc5 (sic). More commits to come...

**Comment by dberzano** [comment]
I have managed to successfully build Run2 and Run3 software with this recipe. Note that:

* I added Python as a dependency for Run2 software as well
* I have fixed Python's recipe in order to _really_ pick our OpenSSL version

**Comment by dberzano** [comment]
I would merge it if there's no objection so that the next O2 nightly will be based on it.

**Pull request merged**